File: mirai-core/src/main/java/net/mamoe/mirai/message/Face.java
package net.mamoe.mirai.message;

/**
 * @author Him188moe
 */
public class Face extends Message {
    private final FaceID id;

    public Face(FaceID id) {
        this.id = id;
    }

    public FaceID getId() {
        return id;
    }

    @Override
    public String toString() {
        // TODO: 2019/9/1
        throw new UnsupportedOperationException();
    }
}

File: mirai-core/src/commonMain/kotlin/net.mamoe.mirai/message/Face.kt
package net.mamoe.mirai.message

/**
 * QQ 自带表情
 */
inline class Face(val id: FaceId) : Message {
    override val stringValue: String get() = "[face${id.value}]"
    override fun toString(): String = stringValue

    companion object Key : Message.Key<Face>
}